src/build/arg/mod.rs

```diff
@@ -2071,6 +2071,46 @@ impl<'help> Arg<'help> {
         self.takes_value(true).multiple_values(true)
     }
 
+    /// Specifies that option values that are invalid UTF-8 should *not* be treated as an error.
+    ///
+    /// **NOTE:** Using argument values with invalid UTF-8 code points requires using
+    /// [`ArgMatches::value_of_os`], [`ArgMatches::values_of_os`], [`ArgMatches::value_of_lossy`],
+    /// or [`ArgMatches::values_of_lossy`] for those particular arguments which may contain invalid
+    /// UTF-8 values.
+    ///
+    /// **NOTE:** Setting this requires [`ArgSettings::TakesValue`]
```

Comment:
Need a debug assert for this at https://github.com/clap-rs/clap/blob/master/src/build/arg/debug_asserts.rs#L89

Reply:
However we infer `takes_value` for positionals means it doesn't work with the debug asserts and adding the debug assert will require explicitly stating `takes_value`.  Unless you have a way around this, I recommend deferring this until https://github.com/clap-rs/clap/discussions/2674

Reply:
Hmm.. we need to modify [checker](https://github.com/clap-rs/clap/blob/412b71efb64d9ac88667b6206c4110460c8d93e1/src/build/arg/debug_asserts.rs#L61) to allow both scenarios.

Reply:
I am working on changing it now. You can ignore this comment chain for this PR.

Reply:
Actually, can you add a line for this in those `assert_arg_flags` checker? I will fix the issue you are talking about in a separate PR.

Reply:
> It causes tests to fail

Ah, I expected that. Nevermind then. I will just create an issue.

Reply:
Check was added in the latest update, unless I'm missing something
https://github.com/clap-rs/clap/pull/2677/files#diff-fa061b40d08726f1fd99f39ea9c37f2ee10fc6365097fdb9a4a4ab4fd2049d23

Reply:
No, I was just updating the comment chain here for future reference.